pkg/java/jar.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2021 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright (c) 2021 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package java
+
+import (
+	"archive/zip"
+	md52 "crypto/md5"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+)
+
+type ClassHash struct {
+	CompleteHash            string
+	BytecodeInstructionHash string
+}
+
+var emptyHash = ClassHash{
+	CompleteHash:            "",
+	BytecodeInstructionHash: "",
+}
```

Comment:
This it the zero-value for ClassHash, no need to intialise it unless you really want it there for semantic value.
`ClassHash{}` can be returned within `HashClass` wherever `empryHash` is used.

Reply:
Purely due to being unfamiliar with go, fixed.